overlord/snapstate/snapmgr.go

```diff
@@ -77,9 +77,17 @@ type SnapSetup struct {
 
 	DownloadInfo *snap.DownloadInfo `json:"download-info,omitempty"`
 	SideInfo     *snap.SideInfo     `json:"side-info,omitempty"`
+
+	// InstanceKey is set by the user during installation and differs for
+	// each instance of given snap
+	InstanceKey string `json:"instance-key,omitempty"`
```

Comment:
there is no non-test code setting this yet, is that right?

Reply:
Correct. The non-test code that makes use of InstanceKey in snapsup/snapst is currently only here https://github.com/bboozzoo/snapd/commits/bboozzoo/parallel-install-overlord